src/wrappers.jl

```diff
@@ -0,0 +1,442 @@
+                    elseif $nesting === 3
+                        all(child) do child2
+                            child2 isa AbstractArray && all(child3 -> geomtrait(child3) isa PointTrait, child2)
+                        end || _parent_type_error(geom)
+                        Z1 = isnothing(Z) ? is3d(first(first(child))) : Z
+                        M1 = isnothing(M) ? ismeasured(first(first(child))) : M
+                        childtype = $childtype
+                        newgeom = childtype.(geom)
+                        return $geomtype{Z1,M1,typeof(newgeom),E}(newgeom, extent)
+                    else
+                        # Otherwise compain the nested child type is wrong
+                        _wrong_child_error($geomtype, $child_trait, child)
+                    end
+                else
+                    # Otherwise compain the nested child type is wrong
+                    _wrong_child_error($geomtype, $child_trait, child)
+                end
+            end
+        else
+            # Or complain the parent type is wrong
+            _parent_type_error(geom)
+        end
+    end
+end
+
+@noinline _wrong_child_error(geomtype, C, child) = throw(ArgumentError("$geomtype must have child objects with trait $C, got $(typeof(child)) with trait $(geomtrait(child))"))
+@noinline _argument_error(T, A) = throw(ArgumentError("$T is does not have $A"))
+@noinline _length_error(T, f, x) = throw(ArgumentError("Length of array must be $(f.f) $(f.x) for $T"))
+@noinline _parent_type_error(geom) = throw(ArgumentError("Object $geom is not a geometry or array of child geometries"))
+
+"""
+    Point
+    Point(geom)
+    Point{Z,M}(geom)
+    Point(; X, Y, [Z, M])
+
+## Arguments
+
+- `geom`: any object returning `PontTrait` from `GeoInterface.trait`.
+
+## Parameters (optional)
+
+These can be used to force points to be interpreted with
+measures or z dimension.
+
+- `Z`: `true` or `false` if there is a z dimension.
+- `M`: `true` or `false` if there are measures.
+
+# Example
+
+```jldoctest
+using GeoInterface
+using GeoInterface.Wrappers
+point = Point{false,true}([1, 2, 3])
+@assert GeoInterface.ismeasured(point) == true
+@assert GeoInterface.is3d(point) == false
+GeoInterface.m(point)
+# output
+3
+```
+"""
+struct Point{Z,M,T} <: WrapperGeometry{Z,M,T}
+    geom::T
+end
+function Point{Z,M}(geom::T) where {Z,M,T}
+    expected_coords = 2 + Z + M
+    ncoord(geom) == expected_coords || _coord_length_error(Z, M, ncoord(geom))
+    Point{Z,M,T}(geom)
+end
+Point(x::Real, y::Real, args::Real...) = Point((x, y, args...))
+Point{Z,M}(x::Real, y::Real, args::Real...) where {Z,M} = Point{Z,M}((x, y, args...))
+function Point(; X::Real, Y::Real, Z::Union{Real,Nothing}=nothing, M::Union{Real,Nothing}=nothing)
+    p = (; X, Y)
+    if !isnothing(Z)
+        p = merge(p, (; Z))
+    end
+    if !isnothing(M)
+        p = merge(p, (; M))
+    end
+    return Point(p)
+end
+function Point(geom)
```

Comment:
Maybe catch the case where we can pass a Point to a Point and prevent recursive wrappers?